AutoComplete/customIndex.js

```diff
@@ -0,0 +1,98 @@
+        return nameArray;
+      },
+      //Searches the returned data based off of the name property.
+      key: ["name"],
+      //Does not cache search.
+      cache: false
+    },
+    //Returns the query (OPTIONAL, kept in for now to make sure nothing breaks when it is removed.).
+    query: {   
+          //Can be used to change the query by looking for certain values                            
+          manipulate: (query) => {
+            //Returns the query
+            return query;
+          }
+    },
+    //Triggers an event which displays the current status of the query in the console log.
+    trigger: {
+      //Trigger runs when searchbox is selected, clicked out of, or has been typed in.
+        event: ["input", "focusin", "focusout"]
+        
+      },
+    //Sorts values in descending order.
+    sort: (a, b) => {                    
+        if (a.match < b.match) return -1;
+        if (a.match > b.match) return 1;
+        return 0;
+    },
+    //Value displayed in searchbox before any value is entered.
+    placeHolder: "Nomenclature", 
+    //ID of input/searchbox which will be used.    
+    selector: "#autoComplete",       
+    //Does not display autocomplete results list until the characters entered into the searchbox exceed the threshold.   
+    threshold: 1,       
+    //Displays search with strict (starts with) results and not loose (contains) results.                                     
+    searchEngine: "strict",  
+    //Defines attributes for the results list.            
+    resultsList: {
+        //Renders/displays the results list.
+        render: true,   
+        //Edits containter id of the results list to be #autoComplete_list.                  
+        container: source => {
+            //Sets id to #autoComplete_list.
+            source.setAttribute("id", "#autoComplete_list");
+        },
+        //Selects the input which the results list will be using.
+        destination: document.querySelector("#autoComplete"),
+        //Sets location of results list relative to the destination (above), can be afterend, afterbegin, beforebegin, and beforeend.
+        position: "afterend",
+        //Sets element type of the results list (ul, div, span, etc...).
+        element: "ul",
+    },
+    //Sets the max number of results that can be returned in a results list for a query (10).
+    maxResults: 10,    
+    //Highlights query string characters inside of the results list.                     
+    highlight: true,     
+    //Edits each item on the list.                  
+    resultItem: {                          
+        content: (data, source) => {
+            //Displays list items only when they match the query.
+            source.innerHTML = data.match;
+        },
+        //Displays result items as list items.
+        element: "li"
+    },
+    /*noResults: () => {                     // Action script on noResults      | (Optional)

```

Comment:
We can change this later, but we should implement a noresult function, or delete this code.